packages/llvm/llvm.3.9/files/install.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+make=$1
+libdir=$2
+
+# Create build dir
+mkdir build
+cd build
+
+# Configure
+cmake ..
+#    -DLLVM_BUILD_LLVM_DYLIB:BOOL=ON \
+#    -DLLVM_LINK_LLVM_DYLIB:BOOL=ON \
+#    ..
+
+# Build
+grep "^ocaml_llvm.*/fast:$" Makefile | tr ":\n" " " | xargs "$make"
```

Comment:
> They have to be located in a separate directory (something like ${prefix}/lib/llvm_{arch}) or be merged to the main META file as a sub package (it would be a nicer solution I think).

> The installation problem: cmake -P bindings/ocaml/cmake_install.cmake installs all the files in ${prefix}/lib/ocaml instead of ${prefix}/lib/llvm

These two problems are actually related (and maybe aren't a problem at all). The idea is that LLVM installs itself into the system library directory, and I think this is actually the layout that's supposed to have, in a distro such as Debian. I think you could write a .install file that provides uninstallation capability from `lib/ocaml`, and put that into `lib/llvm`. Also, `META.foo` is a supported way to provide a META file for package foo in `lib/ocaml` (as documented [here](http://projects.camlcity.org/projects/dl/findlib-1.6.2/doc/ref-html/r721.html)).

> The linking problem: well, it doesn't seems like llvm_libs contains LLVM-3.9 with the BUILD_DYNLIB/LINK_DYNLIB flags. But maybe I'm missing something here.

I'm not sure. I don't have time to dig into that right now. Try #llvm on irc.oftc.net, there are many people who understand how linking works.